components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java

```diff
@@ -149,6 +155,18 @@ public void handle(List<String> roles, String subject, Map<String, String> attri
                  */
                 IdentityUtil.threadLocalProperties.get().put(FrameworkConstants.JIT_PROVISIONING_FLOW, true);
                 if (!userClaims.isEmpty()) {
+
+                    List<Claim> existingUserClaimList = new ArrayList<>(Arrays.asList(userStoreManager
+                            .getUserClaimValues(UserCoreUtil.removeDomainFromName(username), null)));
+                    Set<String> notToDelete = getNotToDeleteClaims();
+                    existingUserClaimList.removeIf(claim -> claim.getClaimUri().contains("/identity/") ||
+                            notToDelete.contains(claim.getClaimUri()) || userClaims.containsKey(claim.getClaimUri()));
+
+                    List<String> toBeDeletedUserClaims = prepareToBeDeletedClaimMappings(attributes);
+                    for (Claim claim : existingUserClaimList) {
+                        toBeDeletedUserClaims.add(claim.getClaimUri());
+                    }
```

Comment:
Can't we use the `addAll` method in `List` in here?

Reply:
As we need to add the claimUri by using `claim.getClaimUri()` method, we cannot use `addAll` method.